File: 225/sol.py
import queue

class MyStack:

    def __init__(self):
        self.q = []

    def push(self, x: int) -> None:
        self.q.append(x)

    def pop(self) -> int:
        element = None
        for i in range(len(self.q) - 1):
            element = self.q.pop(0)
            self.push(element)
        return self.q.pop(0)

    def top(self) -> int:
        element = None
        for i in range(len(self.q)):
            element = self.q.pop(0)
            self.push(element)
        return element

    def empty(self) -> bool:
        if len(self.q) == 0:
            return True
        return False


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 155/sol.go
type MinStack struct {
	s, m []int
}

func Constructor() MinStack {
	return MinStack{make([]int, 0), make([]int, 0)}
}

func (this *MinStack) Push(val int) {
	this.s = append(this.s, val)
	if len(this.m) == 0 || val < this.GetMin() {
		this.m = append(this.m, val)
	} else {
		this.m = append(this.m, this.GetMin())
	}
}

func (this *MinStack) Pop() {
	this.s = this.s[:len(this.s)-1]
	this.m = this.m[:len(this.m)-1]
}

func (this *MinStack) Top() int {
	return this.s[len(this.s)-1]
}

func (this *MinStack) GetMin() int {
	return this.m[len(this.m)-1]
}

/**
 * Your MinStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(val);
 * obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.GetMin();
 */
